codec-http/src/main/java/io/netty/handler/codec/http/CookieDecoder.java

```diff
@@ -25,28 +31,58 @@
 import java.util.TreeSet;
 
 /**
+ * @deprecated Use {@link io.netty.handler.codec.http.cookie.ClientCookieDecoder}
+ * or {@link io.netty.handler.codec.http.cookie.ServerCookieDecoder} instead.
+ *
  * Decodes an HTTP header value into {@link Cookie}s.  This decoder can decode
  * the HTTP cookie version 0, 1, and 2.
  *
  * <pre>
  * {@link HttpRequest} req = ...;
  * String value = req.getHeader("Cookie");
- * Set&lt;{@link Cookie}&gt; cookies = new {@link CookieDecoder}().decode(value);
+ * Set&lt;{@link Cookie}&gt; cookies = {@link CookieDecoder}.decode(value);
  * </pre>
  *
- * @see ClientCookieEncoder
- * @see ServerCookieEncoder
+ * @see io.netty.handler.codec.http.cookie.ClientCookieDecoder
+ * @see io.netty.handler.codec.http.cookie.ServerCookieDecoder
  */
+@Deprecated
 public final class CookieDecoder {
 
+    private final InternalLogger logger = InternalLoggerFactory.getInstance(getClass());
+
+    private static final CookieDecoder STRICT = new CookieDecoder(true);
+
+    private static final CookieDecoder LAX = new CookieDecoder(false);
+
+    private static final String COMMENT = "Comment";
+
+    private static final String COMMENTURL = "CommentURL";
+
+    private static final String DISCARD = "Discard";
+
+    private static final String PORT = "Port";
+
+    private static final String VERSION = "Version";
+
     private static final char COMMA = ',';
 
+    private final boolean strict;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename field "strict" to prevent any misunderstanding/clash with field "STRICT" defined on line 54. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1845)